src/core/modules/impl/handler/JsHandlerApi.ts

```diff
@@ -92,6 +112,17 @@ export class JsHandlerApi<State> extends AbstractContractHandler<State> {
       throwErrorWithName('ConstructorError', templateErrorMessage('refreshState'));
     this.swGlobal.contracts.viewContractState = () =>
       throwErrorWithName('ConstructorError', templateErrorMessage('viewContractState'));
+
+    const disabledVrf = new Proxy(this.swGlobal.vrf, {
```

Comment:
I like the Proxy idea, we could probably use it also for implementing https://github.com/warp-contracts/warp/issues/262

Reply:
Yep, however i would consider implementing different smartWeaveGlobal for different contexts, to be more declarative it could be hard to trace proxies in code.